internal/controller/isbservicerollout_controller_test.go

```diff
@@ -31,60 +32,133 @@ import (
 )
 
 var _ = Describe("ISBServiceRollout Controller", func() {
-	Context("When reconciling a resource", func() {
-		const resourceName = "test-resource"
-
-		ctx := context.Background()
-
-		typeNamespacedName := types.NamespacedName{
-			Name:      resourceName,
-			Namespace: "default", // TODO(user):Modify as needed
-		}
-		isbservicerollout := &apiv1.ISBServiceRollout{}
-
-		BeforeEach(func() {
-			By("creating the custom resource for the Kind ISBServiceRollout")
-			err := k8sClient.Get(ctx, typeNamespacedName, isbservicerollout)
-			if err != nil && errors.IsNotFound(err) {
-				resource := &apiv1.ISBServiceRollout{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      resourceName,
-						Namespace: "default",
-					},
-					// TODO(user): Specify other spec details if needed.
-					Spec: apiv1.ISBServiceRolloutSpec{
-						InterStepBufferService: runtime.RawExtension{
-							Raw: []byte(`{"field":"val"}`),
-						},
-					},
-				}
-				Expect(k8sClient.Create(ctx, resource)).To(Succeed())
+	const (
+		namespace             = "default"
+		isbServiceRolloutName = "isbservicerollout-test"
+		timeout               = 10 * time.Second
+		interval              = 250 * time.Millisecond
+	)
+
+	ctx := context.Background()
+
+	isbServiceRollout := &apiv1.ISBServiceRollout{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: namespace,
+			Name:      isbServiceRolloutName,
+		},
+		Spec: apiv1.ISBServiceRolloutSpec{
+			InterStepBufferService: runtime.RawExtension{
+				Raw: []byte(`{
+									"apiVersion": "v1",
+									"kind": "Service",
+									"metadata": {
+											"name": "my-service"
+									},
+									"spec": {
+											"selector": {
+													"app": "MyApp"
+											},
+											"ports": [{
+													"protocol": "TCP",
+													"port": 80,
+													"targetPort": 9376
+											}]
+									}
+							}`),
+			},
+		},
+	}
+
+	resourceLookupKey := types.NamespacedName{Name: isbServiceRolloutName, Namespace: namespace}
+
+	Context("When applying a ISBServiceRollout spec", func() {
+		It("Should create the ISBServiceRollout if it does not exist", func() {
+			Expect(k8sClient.Create(ctx, isbServiceRollout)).Should(Succeed())
+
+			createdResource := &apiv1.ISBServiceRollout{}
+			Eventually(func() bool {
+				err := k8sClient.Get(ctx, resourceLookupKey, createdResource)
+				return err == nil
+			}, timeout, interval).Should(BeTrue())
+
+			By("Verifying the content of the ISBServiceRollout spec field")
+			Expect(createdResource.Spec).Should(Equal(isbServiceRollout.Spec))
```

Comment:
Also check the created child resource to make sure it has the expected content.

Reply:
Similar to:
- https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout_controller_test.go#L127
- https://github.com/numaproj/numaplane/blob/main/internal/controller/pipelinerollout_controller_test.go#L138